### DISCUSSION TITLE
SpiderNet gave-up

### DISCUSSION THREAD

**Discussion by hydroper** [airsdk/Adobe-Runtime-Support#3261] [General]
Here's what the tutorial for creating a [webpage project](https://hydroper.gitbook.io/spidernet/tutorials/creating-a-webpage) looks for a SpiderNet SDK I was thinking of going over (which was aiming to be similiar to AIR and Flex).

![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/110610266/22fc3bde-7f32-438f-89a8-af7e9d480838)

Details:

* The `ButtonShort` class is a shortcut mapping facility, currently limited to keyboard combinations.

I gave up on the idea for the following reasons:

* TypeScript was more designed for React frameworks, so using inheritance across ECMAScript modules like in the ActionScript 3 model seems like very problematic due to the `extends` clause for example, and sometimes circular imports are just annoying too.
* TypeDoc does not support proper event documentation compared to JSDoc (the `@event` tag doesn't function the sameway). In ActionScript 3 the `[Event]` meta-data just rocks up with Apache Flex.

SpiderNet would abstract away Webpack ("ts-loader" and "dotenv-webpack" for example), Electron, and possibly more, and Spark would abstract away several JavaScript APIs and use certain NPM dependencies (such as `focus-lock` for switching focus through the keyboard arrows).

So, in short, I'm not very happy with TypeScript and ECMAScript.  `^_^` Maybe I'll continue FXRazen on my own, not sure, but you know, a compiler, even for a simple language like ActionScript 3, is quite a lot of work, but it'll be rewarding for me.
